src/types.rs

```diff
@@ -1086,6 +1087,129 @@ impl SimpleAsn1Writable for GeneralizedTime {
     }
 }
 
+/// Used for parsing and writing ASN.1 `GENERALIZED TIME` values accepting
+/// fractional seconds value.
+/// See https://github.com/alex/rust-asn1/issues/491 for discussion.
+#[derive(Debug, Clone, PartialEq, PartialOrd, Hash, Eq)]
+pub struct GeneralizedTimeFractional {
```

Comment:
I don't love the name. If we were doing this all over again, I'd say _this_ should be `GeneralizedTime` and the existing one should be `X509GeneralizedTime` or something, but that's probably too disruptive...

Reply:
Agreed -- some others I thought of were `GeneralizedTimeWithFraction` or `GeneralizedTimeFull`, but I think both of those are as bad if not worse.

OTOH given that this crate is pre-1.0, maybe it's OK to do the breaking thing here? If you'd prefer this be `GeneralizedTime` and the other be `X509GeneralizedTime`, I'm happy to own performing that update/rename on the PyCA side. That would then leave other users with a potentially breaking change, though.

Reply:
Rename is completed - so this PR contains a breaking change 🗡️